# ==============================================================================
# REGRESSOR COMMIT DIFF
# ==============================================================================
# File: dom/media/webrtc/libwebrtcglue/VideoConduit.h
# Commit: 7057437ddf19
# Full Hash: 7057437ddf194b62a158ce84345802e8b2bc6ca8
# Author: Andreas Pehrson <apehrson@mozilla.com>
# Date: 2021-11-01 18:17:33
# Regressor Bug: 1654112
# File Overlap Count: 6
# Description:
#   Bug 1654112 - Constify VideoConduit members that are based on prefs. r=ng
#   
#   Differential Revision: https://phabricator.services.mozilla.com/D112848
# ==============================================================================

diff -r 01569f2b1141 -r 7057437ddf19 dom/media/webrtc/libwebrtcglue/VideoConduit.h
--- a/dom/media/webrtc/libwebrtcglue/VideoConduit.h	Tue Apr 06 18:29:30 2021 +0200
+++ b/dom/media/webrtc/libwebrtcglue/VideoConduit.h	Wed Apr 07 09:44:33 2021 +0200
@@ -222,11 +222,10 @@
 
   WebrtcVideoConduit(RefPtr<WebrtcCallWrapper> aCall,
                      nsCOMPtr<nsISerialEventTarget> aStsThread,
-                     std::string aPCHandle);
+                     Options aOptions, std::string aPCHandle);
   virtual ~WebrtcVideoConduit();
 
-  MediaConduitErrorCode InitMain();
-  virtual MediaConduitErrorCode Init();
+  MediaConduitErrorCode Init();
 
   std::vector<uint32_t> GetLocalSSRCs() override;
   bool SetLocalSSRCs(const std::vector<uint32_t>& ssrcs,
@@ -375,34 +374,28 @@
   // Accessed under mMutex.
   unsigned int mSendingFramerate;
 
-  // Written on main thread at creation,
-  // then written or read on any thread under mTransportMonitor.
-  bool mVideoLatencyTestEnable = false;
-
   // Accessed from any thread under mTransportMonitor.
   uint64_t mVideoLatencyAvg = 0;
 
+  const bool mVideoLatencyTestEnable;
+
   // All in bps.
-  // All written on main thread and guarded by mMutex, except for reads on main.
-  int mMinBitrate = 0;
-  int mStartBitrate = 0;
-  int mPrefMaxBitrate = 0;
+  const int mMinBitrate;
+  const int mStartBitrate;
+  const int mPrefMaxBitrate;
+  const int mMinBitrateEstimate;
+
+  // Max bitrate in bps as provided by negotiation. Call thread only.
   int mNegotiatedMaxBitrate = 0;
-  int mMinBitrateEstimate = 0;
 
   // Set to true to force denoising on.
-  // Written at creation, then read anywhere.
-  bool mDenoising = false;
+  const bool mDenoising;
 
   // Set to true to ignore sink wants (scaling due to bwe and cpu usage).
-  // Written at creation, then read anywhere.
-  bool mLockScaling = false;
+  const bool mLockScaling;
 
-  // Written at creation, then read anywhere.
-  uint8_t mSpatialLayers = 1;
-
-  // Written at creation, then read anywhere.
-  uint8_t mTemporalLayers = 1;
+  const uint8_t mSpatialLayers;
+  const uint8_t mTemporalLayers;
 
   static const unsigned int sAlphaNum = 7;
   static const unsigned int sAlphaDen = 8;